documents/api/viewsets.py

```diff
@@ -38,24 +38,71 @@
     CreateAttachmentSerializer,
     DocumentSerializer,
 )
-from ..serializers.document import DocumentMetadataSerializer, GDPRSerializer
+from ..serializers.document import (
+    DocumentMetadataSerializer,
+    DocumentStatisticsSerializer,
+    GDPRSerializer,
+)
 from ..serializers.status_history import StatusHistorySerializer
 from ..utils import get_decrypted_file
 from .docs import (
     attachment_viewset_docs,
     document_gdpr_viewset,
     document_metadata_viewset_docs,
+    document_statistics_viewset_docs,
     document_viewset_docs,
 )
-from .filtersets import DocumentFilterSet, DocumentMetadataFilterSet
+from .filtersets import (
+    DocumentFilterSet,
+    DocumentMetadataFilterSet,
+    DocumentStatisticsFilterSet,
+)
 from .querysets import (
     get_attachment_queryset,
     get_document_gdpr_data_queryset,
     get_document_metadata_queryset,
     get_document_queryset,
+    get_document_statistics_queryset,
 )
 
 
+@extend_schema_view(**document_statistics_viewset_docs)
+class DocumentStatisticsViewSet(AuditLoggingModelViewSet):
+    queryset = Document.objects.none()
+    serializer_class = DocumentStatisticsSerializer
+    filter_backends = [DjangoFilterBackend]
+    filterset_class = DocumentStatisticsFilterSet
+
+    def get_queryset(self):
+        user = self.request.user
```

Comment:
I have said this before but this method does everything else but get_queryset. Well naming is becacuse of DRF but this can be confusing

Reply:
I'll keep this for now, but I'm working on refactoring large parts of the code, to make it easier to follow and understand. Moving querysets to their rightful place is one of the changes I'm making.